php/Blockonomics.php

```diff
@@ -131,59 +131,56 @@ public function check_get_callbacks_response_code($response){
         {
             $error_str = __('You have not entered an xPub', 'blockonomics-bitcoin-payments');
         }
-        elseif (count($response_body) == 1)
+        //if merchant has at least one xPub on his Blockonomics account
+        elseif (count($response_body) >= 1)
         {
-            $response_callback = '';
-            $response_address = '';
-            if(isset($response_body[0])){
-                $response_callback = isset($response_body[0]->callback) ? $response_body[0]->callback : '';
-                $response_address = isset($response_body[0]->address) ? $response_body[0]->address : '';
+            $error_str = $this->examine_server_callback_urls($response_body, $wordpress_callback_url, $api_url, $crypto);
+        }
+        return $error_str;
+    }
+
+    // checks each existing xpub callback URL to update and/or use
+    public function examine_server_callback_urls($response_body, $wordpress_callback_url, $api_url, $crypto)
+    {
+        $base_url = preg_replace('/https?:\/\//', '', $api_url);
+        $available_xpub = '';
+        //Go through all xpubs on the server and examine their callback url
+        foreach($response_body as $one_response){
+            $server_callback_url = isset($one_response->callback) ? $one_response->callback : '';
+            $xpub = isset($one_response->address) ? $one_response->address : '';
+            // No callback
+            if(!$server_callback_url){
+                $available_xpub = $xpub;
+            // Exact match
+            }else if($server_callback_url == $wordpress_callback_url){
+                return '';
             }
-            $response_callback_without_schema = preg_replace('/https?:\/\//', '', $response_callback);
-            if(!$response_callback || $response_callback == null)
-            {
-                //No callback URL set, set one 
-                $this->update_callback($callback_url, $crypto, $response_address);
-            }
-            elseif($response_callback_without_schema != $callback_url_without_schema)
-            {
-                $base_url = get_bloginfo('wpurl');
-                $base_url = preg_replace('/https?:\/\//', '', $base_url);
-                // Check if only secret differs
-                if(strpos($response_callback, $base_url) !== false)
-                {
-                    //Looks like the user regenrated callback by mistake
-                    //Just force Update_callback on server
-                    $this->update_callback($callback_url, $crypto, $response_address);
-                }
-                else
-                {
-                    $error_str = __("You have an existing callback URL", 'blockonomics-bitcoin-payments');
-                }
-                
+            // Partial Match - Only secret or protocol differ
+            else if(strpos($server_callback_url, $api_url) === 0 || strpos($server_callback_url, $base_url)){
```

Comment:
The code is clean and readable. Checking for a Partial Match requires further work as `strpos($server_callback_url, $base_url)` updates the callback URL incorrectly if a different subdomain is used.

Test case (different subdomain):
- Blockonomics Server: http://subdomain.shop.com/wc-api/WC_Gateway_Blockonomics/?secret=123
- WordPress URL: http://shop.com/wc-api/WC_Gateway_Blockonomics/?secret=123

The URL should not update in this case as subdomain.shop.com can be a different store to shop.com

Reply:
You're right, it was fairly easy to fix. I added the a variable `$wp_base_url` and make sure that $base_url start at the same position. That way no subdomain can be "hiding" on the server without getting detected and being called out as a multiple callback error.